nixos/modules/hardware/busylight.nix

```diff
@@ -0,0 +1,25 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
```

Comment:
This needs to be removed. Add the`lib.` prefix instead.

Reply:
Interesting.  For what reason?  I attempted to mimic the style found elsewhere in `nixos/modules/hardware` many of which use `with lib;`.

I'll make the change, however.

Reply:
There are several compelling reasons to avoid using a global `with` statement.

Firstly, employing such a statement imports all the functions from `lib` into your current scope. This can potentially cloud your code's clarity and lead to confusion. A more preferable alternative would be using a targeted statement like `inherit (lib) abc def;` which enhances the readability of your code. This way, you can accurately trace the origin of each function.

Secondly, a global `with` statement could interfere with your Language Server Protocol (LSP). Given that the entire `lib` scope is imported, your LSP might struggle to pinpoint the sources of various functions, resulting in a poor development experience.

There are additional considerations regarding this topic that fall outside my expertise, but I encourage you to post this question on discourse for a more comprehensive discussion.